etsin_finder/frontend/js/components/dataset/formatselect.jsx

```diff
@@ -0,0 +1,237 @@
+{
+  /**
+   * This file is part of the Etsin service
+   *
+   * Copyright 2017-2018 Ministry of Education and Culture, Finland
+   *
+   *
+   * @author    CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
+   * @license   MIT
+   */
+}
+
+
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import translate from 'counterpart'
+import { darken } from 'polished'
+import PropTypes from 'prop-types'
+import Translate from 'react-translate-component'
+import { HelpIcon } from '../qvain/general/form'
+import Tooltip from '../qvain/general/tooltip'
+import Button from '../general/button'
+
+
+const SelectContainer = styled.div`
+  width: ${props => props.width};
+  margin-right: 1em;
+  position: relative;
+`
+
+/* prettier-ignore */
+const ListButton = styled(Button)`
+  color: ${props => props.color};
+  padding: ${props => props.padding};
+  border-radius: 0;
+  width: 100%;
+  text-align: left;
+  display: block;
+  border: 0;
+  background: ${props => props.background};
+  &:hover {
+    background: ${props => (props.removed ? darken(0.1, props.theme.color.error) : darken(0.1, props.background))};
+  }
+`
+
+const ListItem = styled(ListButton)`
+  &:first-of-type {
+    border-top: 1px solid ${props => darken(0.2, props.background)};
+  }
+  &:last-of-type {
+    border-radius: 0 0 5px 5px;
+  }
+  background-color: white;
+  color: ${props => props.color};
+  border: 1px solid;
+`
+
+const Controller = styled(ListButton)`
+  transition: 0.3s ease;
+  border-radius: ${props => (props.isOpen ? '5px 5px 0 0' : '5px 5px 5px 5px')};
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  color: white;
+  &:after {
+    transform: ${props => (props.isOpen ? 'rotate(180deg)' : 'rotate(0deg)')};
+    content: '';
+    width: 0.3em;
+    border-top: 0.3em solid white;
+    border-right: 0.3em solid transparent;
+    border-left: 0.3em solid transparent;
+  }
+`
+
+const InfoPosition = styled.div`{
+  position: absolute;
+  top: 8px; 
+  right: -25px;
+}
+`
+
+export default class FormatSelect extends Component {
+  constructor(props) {
+    super(props)
+
+    this.timeoutID = undefined
+
+    this.state = {
+      isOpen: false,
+      options: props.options,
+      newestColor: props.newestColor ? props.newestColor : props.background,
+      color: props.color,
+      background: props.background,
+      padding: props.padding,
+      width: props.width,
+      tooltipOpen: false,
+    }
+  }
+
+  static getDerivedStateFromProps(newProps, prevState) {
```

Comment:
I think that the `getDerivedStateFromProps` is unnecessary because the component should re-render if the props that the component is called with change. Also, if the values in the state that are taken from the props probably don't need to go to the state, they could be read directly from the props.

the state could look like this(suggestion, haven't tried):
```js
this.state = {
      isOpen: false,
      tooltipOpen: false,
    }
// the removed values could be used like this:
width={this.props.width}
// instead of writing it to the state and
width={this.state.width}

```